lib/tests/streamlit/select_slider_test.py

```diff
@@ -95,6 +125,66 @@ def test_format_func(self):
         self.assertEqual(c.default, [1])
         self.assertEqual(c.options, DAYS_OF_WEEK)
 
+    def test_numpy_array_no_value(self):
+        """Test case when options is numpy array without value"""
+        st.select_slider("the label", options=np.array([1, 2, 3, 4]))
+
+        c = self.get_delta_from_queue().new_element.slider
+        self.assertEqual(c.default, [0])
+
+    def test_numpy_array_with_value(self):
+        """Test case when options is numpy array"""
+        st.select_slider("the label", value=3, options=np.array([1, 2, 3, 4]))
+
+        c = self.get_delta_from_queue().new_element.slider
+        self.assertEqual(c.default, [2])
+
+    def test_numpy_array_with_range(self):
+        """Test case when options is numpy array and value is range"""
+        st.select_slider(
+            "the label", value=(2, 5), options=np.array([1, 2, 3, 4, 5, 6])
+        )
+
+        c = self.get_delta_from_queue().new_element.slider
+        self.assertEqual(c.default, [1, 4])
+
+    def test_numpy_array_with_invalid_value(self):
+        """Test case when value is not in numpy array"""
+        with pytest.raises(ValueError) as exc_message:
```

Comment:
nit: can drop the `as exc_message` if it's not used below (or add a check for its error message, but I don't think the error message is particularly interesting)

Same comment below on L183